app.py

```diff
@@ -185,17 +186,37 @@ def set_frame_range(self, engine, in_frame, out_frame):
 
         elif engine == "tk-nuke":
             import nuke
-
+            root = nuke.root()
+            lock_range = root["lock_range"]
             # unlock
-            locked = nuke.root()["lock_range"].value()
+            locked = lock_range.value()
+            change_msg = 'from %s-%s to %s-%s' % (prev_in, prev_out, in_frame, out_frame)
             if locked:
-                nuke.root()["lock_range"].setValue(False)
+                ret = nuke.ask("Frame range is currently locked,\n"
+                               "OK to update frame range %s ?" % change_msg)
+                if not ret:
+                    return
+            else:
+                p = nuke.Panel("Shall lock ?")
```

Comment:
I would prefer if you keep using Qt stuff directly and not Nuke stuff. Using Qt stuff will make it clear that it comes from this app and Tank as Tank applies some styling to widgets/dialogs, ansd so we can easily know the pop-up is coming from Tank apps/engines.